core/src/main/java/org/elasticsearch/search/profile/SearchProfileShardResults.java

```diff
@@ -75,26 +87,71 @@ public void writeTo(StreamOutput out) throws IOException {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject("profile").startArray("shards");
-
-        for (Map.Entry<String, ProfileShardResult> entry : shardResults.entrySet()) {
+        builder.startObject(PROFILE_FIELD).startArray(SHARDS_FIELD);
+        TreeSet<String> sortedKeys = new TreeSet<>(shardResults.keySet());
```

Comment:
it is not just for tests. Unfortunately this object uses a map to store information that goes into a json array, which is by definition ordered. Either we change the format of the response or we guarantee ordering then. I think this is a good way to guarantee ordering and make the response more human readable too.